### PR TITLE
feat: [CDS-40590]: Added PDC infra runtime/Inputset form support

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
### Summary

> Added runtime/inputSet form support for PDC infra
> Added validations

#### Screenshots

https://user-images.githubusercontent.com/98325173/183982512-bed79b9c-4ba1-4786-bbaf-ec1a0caf862a.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 28 Issues 
* Code issue 28 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10478)

**Comment by Sagar-Rajput7** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureInterface.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+export interface PDCInfrastructureUI {
+  hostsType?: number
+  allowSimultaneousDeployments?: boolean
+  attributeFilters?: string
+  hosts: string
+  connectorRef?: string
+  delegateSelectors?: string[]
+  hostFilters: string
+  sshKey?: SecretReferenceInterface | void
```
Comment (vikrant-harness) [inline src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureInterface.tsx:35]: Do we need both sshKey and credentialsRef

File: src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpec.tsx [hunk 2]
```diff
@@ -192,7 +152,9 @@ const PDCInfrastructureSpecEditable: React.FC<PDCInfrastructureSpecEditableProps
         ...initialValues,
         hosts: initialValues.hosts ? initialValues.hosts.join('\n') : '',
         hostFilters: initialValues.hostFilters ? initialValues.hostFilters.join('\n') : '',
-        attributeFilters: initialValues.attributeFilters
+        attributeFilters: isString(initialValues.attributeFilters)
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpec.tsx:155]: Why do we need this condition if we are setting the same variable in both cases?

Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpec.tsx:155]: Move it to a function to improve readability

File: src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpecInputForm.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+              if (getMultiTypeFromValue(template?.connectorRef) === MultiTypeInputType.RUNTIME) {
+                data.connectorRef = value.connectorRef
+              }
+              if (getMultiTypeFromValue(template?.attributeFilters) === MultiTypeInputType.RUNTIME) {
+                data.attributeFilters = parseAttributes(value.attributeFilters || '')
+              }
+              if (getMultiTypeFromValue(template?.credentialsRef) === MultiTypeInputType.RUNTIME) {
+                data.credentialsRef = (value.credentialsRef || value.sshKey || '') as string
+              }
+              delayedOnUpdate(data)
```
Comment (vikrant-harness) [inline src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpecInputForm.tsx:115]: if handling through formik , can we reduce the delay of 1000?

File: src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpecInputForm.tsx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    connectorRef: ''
+  }
+
+  useEffect(() => {
+    const setInitial = async () => {
+      const values = {
+        ...initialValues,
+        hosts: initialValues.hosts ? initialValues.hosts.join('\n') : '',
+        hostFilters: initialValues.hostFilters ? initialValues.hostFilters.join('\n') : '',
+        attributeFilters: isString(initialValues.attributeFilters)
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpecInputForm.tsx]: Move this to a function

File: src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpecInputForm.tsx [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+                multiTypeProps={{ allowableTypes, expressions }}
+                gitScope={{ repo: repoIdentifier || '', branch, getDefaultFromOtherRepo: true }}
+                setRefValue
+              />
+            </div>
+          )}
+          {getMultiTypeFromValue(template?.hosts) === MultiTypeInputType.RUNTIME && (
+            <div className={cx(stepCss.formGroup, stepCss.md)}>
+              <FormMultiTypeTextAreaField
+                key="hosts"
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpecInputForm.tsx]: Use Strings

File: src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpecInputForm.tsx [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+                  expressions,
+                  allowableTypes
+                }}
+              />
+            </div>
+          )}
+          {getMultiTypeFromValue(template?.hostFilters) === MultiTypeInputType.RUNTIME && (
+            <div className={cx(stepCss.formGroup, stepCss.md)}>
+              <FormMultiTypeTextAreaField
+                key="hostFilters"
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpecInputForm.tsx]: Use Strings

File: src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpecInputForm.tsx [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+                  expressions,
+                  allowableTypes
+                }}
+              />
+            </div>
+          )}
+          {getMultiTypeFromValue(template?.attributeFilters) === MultiTypeInputType.RUNTIME && (
+            <div className={cx(stepCss.formGroup, stepCss.md)}>
+              <FormMultiTypeTextAreaField
+                key="attributeFilters"
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpecInputForm.tsx]: use Strings

File: src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpec.tsx [hunk 8]
```diff
@@ -414,21 +376,30 @@ const PDCInfrastructureSpecEditable: React.FC<PDCInfrastructureSpecEditableProps
             initialValues={formikInitialValues}
             validationSchema={getValidationSchema(getString) as Partial<PDCInfrastructureUI>}
             validate={value => {
-              const data: Partial<PdcInfrastructure> = {
+              const data: Partial<PDCInfrastructureYAML> = {
                 allowSimultaneousDeployments: value.allowSimultaneousDeployments,
                 delegateSelectors: value.delegateSelectors,
                 sshKey: value.sshKey,
```
Comment (vikrant-harness) [inline src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpec.tsx:382]: here too do we need both? seems to perform the same action

**Review by vikrant-harness — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

**Pull request merged**